Aggiunto script per la creazione di AppImage.

demperor-music opened the pull request:
Risolve #11. 
Lo script compila il programma, scarica tutte le dipendenze e le impacchetta in un singolo file AppImage.
Lo script va eseguito nella cartella "Installer/SpaghettiCh2".

L'AppImage è stato testato su Arch, Fedora ed Alpine.

Per modificare il modo in cui l'app appare nel menù delle app se l'utente decide di installarla, modificare la sezione che inizia con "Creazione file .desktop..."

<img width="510" height="296" alt="image" src="https://github.com/user-attachments/assets/e276eef6-7a25-4bfc-b801-fb31c741b10c" />
<img width="1070" height="1186" alt="image" src="https://github.com/user-attachments/assets/2fd1baee-f5d0-4e1f-b2bd-edc61277b972" />

demperor-music committed:
Aggiunto script per la creazione di AppImage.

Questo dovrebbe servire per prevenire problemi di compatibilità futuri con le versioni di Linux.
Lo script compila il programma, scarica tutte le dipendenze e le impacchetta in un singolo file AppImage.
Il file risultante è stato testato su Arch, Fedora ed Alpine.

jcb936 commented:
Nice - gli diamo un'occhiata e la testiamo!

jcb936 commented on SpaghettiCh2/linuxBuild.sh:
```diff
@@ -0,0 +1,138 @@
+cd $current_dir
+
+echo -n "Compilazione del patcher..."
+dotnet build -c Release -o $_pkgdir/opt/spaghettiproject
```
Qui andrebbe usato il comando di .net per deploy il progetto properly:

```
dotnet restore USPInstaller.csproj -r linux-x64
dotnet publish USPInstaller.csproj -r linux-x64 --self-contained
```

demperor-music committed:
Update linuxBuild.sh

Added quotes around variable names for safety

demperor-music committed:
Update linuxBuild.sh

Something broke in the last commit, so i fixed it.
(a variable name somehow got cut out)

jcb936 commented on SpaghettiCh2/linuxBuild.sh:
```diff
@@ -0,0 +1,138 @@
+echo -n "Creazione script di avvio..."
+echo "#!/bin/bash
+
+\$APPDIR/usr/bin/dotnet \$APPDIR/opt/spaghettiproject/USPInstaller.dll" > "$_pkgdir/usr/bin/USPInstaller.sh"
```
Senza dover creare uno script di avvio - se cambi la generazione dell'eseguibile a `dotnet publish --self-contained ` come da commento sopra hai tutto self contained per far partire tutto, senza dover anche avere il runtime e puoi runnare ./USPInstaller direttamente
